src/Libraries/RevitNodes/Elements/Room.cs

```diff
@@ -0,0 +1,219 @@
+        /// Set internal element
+        /// </summary>
+        private void InitElement(Autodesk.Revit.DB.Architecture.Room element)
+        {
+            InternalSetElement(element);
+        }
+
+        private void Init(Autodesk.Revit.DB.Level level, Autodesk.Revit.DB.XYZ location, string name, string number)
+        {
+            // Get document and open transaction
+            Autodesk.Revit.DB.Document document = DocumentManager.Instance.CurrentDBDocument;
+            TransactionManager.Instance.EnsureInTransaction(document);
+
+            // Get existing room element if possible
+            var RoomElem = ElementBinder.GetElementFromTrace<Autodesk.Revit.DB.Architecture.Room>(document);
+
+            if (RoomElem == null)
+            {
+                // Create new Room element
+                RoomElem = document.Create.NewRoom(level, new Autodesk.Revit.DB.UV(location.X, location.Y));
+            }
+            else
+            {
+                // Update Location only
+                Autodesk.Revit.DB.LocationPoint point = (Autodesk.Revit.DB.LocationPoint)RoomElem.Location;
+                point.Point = location;
+            }
+
+            // Apply name and number if set
+            if (name != string.Empty)
+            {
+                RoomElem.Name = name;
+            }
+
+            if (number != string.Empty)
+            {
+                RoomElem.Number = number;
+            }
+
+            InternalSetElement(RoomElem);
+
+            // Commit transaction
+            TransactionManager.Instance.TransactionTaskDone();
+
+            ElementBinder.SetElementForTrace(this.InternalElement);
+        }
+
+        #endregion
+
+        #region Private mutators
+
+        /// <summary>
+        /// Set the internal Element, ElementId, and UniqueId
+        /// </summary>
+        private void InternalSetElement(Autodesk.Revit.DB.Architecture.Room element)
+        {
+            InternalRevitElement = element;
+            InternalElementId = element.Id;
+            InternalUniqueId = element.UniqueId;
+        }
+
+        #endregion
+
+        #region Public static constructors
+
+        /// <summary>
+        /// Create a Revit Room Element
+        /// </summary>
+        /// <param name="level">Level the room is hosted on</param>
+        /// <param name="location">Location for the center of the room</param>
+        /// <param name="name">Room name</param>
+        /// <param name="number">Room number</param>
+        /// <returns></returns>
+        public static Room ByLocation(Elements.Level level, Autodesk.DesignScript.Geometry.Point location, string name = "", string number = "")
+        {
+            return new Room((Autodesk.Revit.DB.Level)level.InternalElement, location.ToRevitType(true), name, number);
+        }
+
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Get room name
+        /// </summary>
+        public string GetName
```

Comment:
Property name should not include Get. Can we have Name, Number,Area,Height etc. as property?